### ISSUE TITLE
M532 - Jammed state not handled properly

### ISSUE THREAD

**Issue by clearwave1** [formatBCE/Airbnk-MQTTOpenGateway#15]
I have a new M532 that is setup and working well most of the time with your ESPHome firmware and the HA integration by @rospogrigio .

Once every few days, the lock goes into a "Jammed" state which is shown by the WeHere app. When this happens, I can still request an unlock from the WeHere app and it is successful. This also resets the state back to unlocked. However, any request through HA and your firmware does not work even though the logs show it was sent successfully. This is also true even though the WeHere app fixed the state of the lock. As an aside, I am using a separate account in WeHere from that which is used in the HA integration so there should be no conflict.

Initially, the HA integration would prevent the request when in this state but I removed the condition from the code and it will now let the call go through to the firmware. I did this both for testing and because it sometimes works to unlock the door.

Now the really strange part - once the lock cannot be opened by your firmware it takes a full HA restart to fix it.
Here are some things I've tried that do not work:
- Reload the firmware
- Reset the ESP32
- Restart ESPHome

I would paste the logs here but they will be of no use as they are exactly the same when it succeeds and when it fails.

Let me know your thoughts and if there are other things I can try to test or code changes that can be tried in the firmware code.

**Comment by formatBCE** [comment]
Hi!
This "lock" happens, because these locks use some bits in payload, that are indicating, in simple words, ever-rising number of operations, done on the lock. This number received on initialization of integration, once. And while official app has its own methods to "fix" that number, the rospogrigio's integration can't (because basically after one incorrect send there's no way to receive correct number).
So it "assumes" that number, and adds 1 each time on command send. 
Integration reload would help, yes.

**Comment by clearwave1** [comment]
Thanks for your quick response and the information you provided.

I wonder if an HA reload of the integration without an HA full restart would also get the number. Do you think this would work?

If so, I can setup an automation that checks the state for "Jammed" and request an integration reload when that happens.

Do you think this is possible and will it help?

**Comment by formatBCE** [comment]
> Thanks for your quick response and the information you provided.
> 
> I wonder if an HA reload of the integration without an HA full restart would also get the number. Do you think this would work?
> 
> If so, I can setup an automation that checks the state for "Jammed" and request an integration reload when that happens.
> 
> Do you think this is possible and will it help?

I really don't know if this integration can be reloaded (don't remember, don't use it for long) - but yes it would help for sure. :)
ESP is dumb, it's just bridge from MQTT to Bluetooth. The real state is there, in integration.

**Comment by clearwave1** [comment]
I had the lock jam tonight so I tried reloading the integration and the unlock command started working again.

I noticed from the event counter entity that it is updated from the lock when the integration is loaded so if it gets out of sync with the lock, a reload will fix  it.

To make it easier, I have added an HA automation to take care of this so my family doesn't need to wait for me to fix it.

Thanks again for your suggestions - they helped me solve this.

NOTE: You might want to add this to the documentation in case others have this problem.
